System/StandardIncludes.h

```diff
@@ -29,6 +29,7 @@
 #include <cerrno>
 #include <cassert>
 #include <limits>
+#include <experimental/filesystem>
```

Comment:
Filesystem routines.
It was merged into the C++17 standard and is just `std::filesystem` there but enabling ++17  breaks a bunch of other stuff so i'm using the not-ISO one instead.

Need this to not rely on Allegro for loading and unzipping of files.
It's currently used only to find that a filename in a path exists because for god knows what reason there aren't any convenient methods for this except in here.
I can revert to using the Allegro routines for this too but I prefer not to (for the future).
Alternatively, Win32 API routines can be used for this.

Reply:
Okay, I'm fine with leaving it as is but can you put either an issue or a TODO comment to come back to this in future (e.g. when we upgrade c++ or allegro)? It sounds a little messy and fragile and I'd like to clean up things like that over time.